### PR TITLE
Implement Krnlmon CI pipeline

### PR DESCRIPTION
**Pull request by ffoulkes**
- Define a GitHub pipeline to run the DPDK unit test suite when a push or pull request is issued against the main branch of the krnlmon repository.

### CODE REVIEW

**Review by 5abeel — APPROVED** [review]
lgtm

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  # If workflow for PR or push is already running, stop it and start a new one.
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  #---------------------------------------------------------------------
+  # Environment variables
+  #---------------------------------------------------------------------
+  SDE_REPOSITORY: ipdk-io/p4dev.dpdk-sde
+  SDE_TAG: 2023.10.1
```
Comment (nupuruttarwar) [inline .github/workflows/pipeline.yml:21]: What does this tag signify?  Where can we see details about which SDE SHA is being picked?

Comment (ffoulkes) [reply .github/workflows/pipeline.yml:21]: The answers to your questions may be found here:

- [dpdk-sde 2023.10.1](https://github.com/ipdk-io/p4dev.dpdk-sde/releases/tag/2023.10.1)
- [stratum-deps v1.2.1](https://github.com/ipdk-io/stratum-deps/releases/tag/v1.2.1)


I've scheduled a CI Pipeline Review (Teams meeting) for 9:00 am Wednesday to discuss the mechanics of the revised pipeline. I figured a guided tour would be easier than a written explanation.

**Review by n-sandeep — APPROVED** [review]
LGTM

**Pull request merged**